cmd/lncli/commands.go

```diff
@@ -2285,14 +2292,32 @@ func sendPayment(ctx *cli.Context) error {
 
 	var rHash []byte
 
-	switch {
-	case ctx.IsSet("payment_hash"):
-		rHash, err = hex.DecodeString(ctx.String("payment_hash"))
-	case args.Present():
-		rHash, err = hex.DecodeString(args.First())
-		args = args.Tail()
-	default:
-		return fmt.Errorf("payment hash argument missing")
+	if ctx.Bool("key_send") {
+		if ctx.IsSet("payment_hash") {
```

Comment:
Similarly, error out of the dest isn't set.

Reply:
Is already checked above